motoman_driver/src/industrial_robot_client/joint_trajectory_action.cpp

```diff
@@ -614,9 +637,19 @@ bool JointTrajectoryAction::withinGoalConstraints(
   {
     int last_point = traj.points.size() - 1;
 
+    std::vector<std::string> state_names;
+    std::vector<double> state_values;
+
+    // joint torso_joint_b2 is not in traj.joint_names, so the last group is ignored here.
```

Comment:
See my earlier comment: ignoring the last group is not a good idea on systems that actually do have 4 groups, but do not have the special joint<->group mapping that the SDA series has.